airflow/providers/hashicorp/secrets/vault.py

```diff
@@ -0,0 +1,156 @@
+        backend_kwargs = {"path": "airflow", "url": "http://127.0.0.1:8200"}
+
+    For example, if your keys are under ``airflow/AIRFLOW_CONN_SMTP_DEFAULT``, this would be accessible if you
+    provide ``{"path": "airflow"}`` and request conn_id ``smtp_default``.
+
+    :param path: Specifies the path of the secret to read.
+    :type path: str
+    :param url: Base URL for the Vault instance being addressed.
+    :type url: str
+    :param auth_type: Authentication Type for Vault (one of 'token', 'ldap', 'userpass', 'approle',
+        'github'). Default is ``token``.
+    :type auth_type: str
+    :param mount_point: The "path" the secret engine was mounted on. (Default: ``secret``)
+    :type mount_point: str
+    :param token: Authentication token to include in requests sent to Vault.
+        (for ``token`` and ``github`` auth_type)
+    :type token: str
+    :param username: Username for Authentication (for ``ldap`` and ``userpass`` auth_type)
+    :type username: str
+    :param password: Password for Authentication (for ``ldap`` and ``userpass`` auth_type)
+    :type password: str
+    :param role_id: Role ID for Authentication (for ``approle`` auth_type)
+    :type role_id: str
+    :param secret_id: Secret ID for Authentication (for ``approle`` auth_type)
+    :type secret_id: str
+    """
+    def __init__(
+        self,
+        path: str,
+        url: Optional[str] = None,
+        auth_type: str = 'token',
+        mount_point: str = 'secret',
+        token: Optional[str] = None,
+        username: Optional[str] = None,
+        password: Optional[str] = None,
+        role_id: Optional[str] = None,
+        secret_id: Optional[str] = None,
+        **kwargs
+    ):
+        super().__init__()
+        self.path = path.rstrip('/')
+        self.url = url
+        self.auth_type = auth_type
+        self.kwargs = kwargs
+        self.token = token
+        self.username = username
+        self.password = password
+        self.role_id = role_id
+        self.secret_id = secret_id
+        self.mount_point = mount_point
+        self._client: Optional[hvac.Client] = None
+
+    def get_client(self) -> hvac.Client:
+        """
+        Return an authenticated Hashicorp Vault client
+        """
+        if not self._client:
+            self._client = hvac.Client(url=self.url, **self.kwargs)
+            if self.auth_type == "token":
+                self._client.token = self.token
+            elif self.auth_type == "ldap":
+                self._client.auth.ldap.login(
+                    username=self.username, password=self.password)
+            elif self.auth_type == "userpass":
+                self._client.auth_userpass(username=self.username, password=self.password)
+            elif self.auth_type == "approle":
+                self._client.auth_approle(role_id=self.role_id, secret_id=self.secret_id)
+            elif self.auth_type == "github":
```

Comment:
Can you also add GCP? This is a very often used cloud provider for Airflow.  We have operators for [31 Google service](https://airflow.readthedocs.io/en/latest/operators-and-hooks-ref.html#id13). The code will look similar to the following.
```
    elif self.auth_type == "gcp":
        credentials = self.get_gcp_credentialss()
        self._client.auth.gcp.configure(credentials=credentials, mount_point=self.mount_point):
    else:



def get_gcp_credentialss():
    if self.gcp_key_path:
        # Get credentials from a JSON file.
        if key_path.endswith('.json'):
            self.log.debug('Getting connection using JSON key file %s', key_path)
            credentials = (
                google.oauth2.service_account.Credentials.from_service_account_file(
                    key_path, scopes=self.scopes)
            )
        elif key_path.endswith('.p12'):
            raise AirflowException(
                'Legacy P12 key file are not supported, use a JSON key file.'
            )
        else:
            raise AirflowException('Unrecognised extension for key file.')
    else:
        self.log.debug(
            'Getting connection using `google.auth.default()` since no key file is defined.'
        )
        credentials, _ = google.auth.default(scopes=self.scopes)

    return credentials
```

Reply:
Sure

Reply:
Updated